image/templates/helm/stackrox-central/internal/config-shape.yaml

```diff
@@ -164,6 +170,9 @@ allowNonstandardNamespace: null # bool
 allowNonstandardReleaseName: null # bool
 allowUnsupportedHelmVersion: null # bool
 enableOpenShiftMonitoring: null # bool
+configAsCode:
```

Comment:
Central has already a `declarativeConfig` section with the same intention of providing configs as code. I found the `declarativeConfig` not optimal but customers are already using it.
Maybe putting there on a feature basis, e.g. `policiesAsCode.enabled`? Eventually more could be added there.

https://github.com/stackrox/stackrox/blob/7d176aaff7da22a5e4ff306362a2dd2ae88d7e72/image/templates/helm/stackrox-central/internal/config-shape.yaml#L74-L77

Reply:
I'm torn on this suggestion.  This (configAsCode) is a distinct feature from declarativeConfig, but both features could certainly be described as "declarative config".  It might be confusing to customers to have both a `declarativeConfig` and `configAsCode` section, but at the same time it might be confusing to mash two distinct features under one section in the values.yaml.

Maybe a different name?

```
declarativeConfig:
  kubeNative:
    policies: true
```

This mirrors the structure of declarativeConfig.

Another consideration is whether to put configAsCode config under the `central` key or not.  I was leaning towards not since it's a separate deployment, following the pattern of scanner.

Reply:
Hm, I agree, declarative config feels not right too.
Is a `policyAsCode` config necessary? Could it just be under `config-controller.enabled` and enables automtically all CRD related stuff? I don't understand why the split between these configs was done.

Reply:
I explained why the CRD has a separate toggle flag [here](https://github.com/stackrox/stackrox/pull/12799#discussion_r1782051317).  For now at least, disabling policy as code also removes a feature flag env var from the central deployment which disables the k8s service account m2m config.